### PR TITLE
Feature/lock nitric version

### LINKED ISSUE #194: Ability to lock versions for stacks and functions
Ability to lock versions for stacks and functions.

Should include validation of semver

```yaml
name: my-stack

version: 0.12.0-rc.15

# Nitric functions
functions:
  create:
    handler: functions/create.ts
    version: 0.12.0-rc.10
  read:
    handler: functions/read.ts
  list:
    handler: functions/list.ts
# Nitric collections
collections:
  examples: {}
# Nitric APIs
apis:
  examples: api.yaml

```

### PR DESCRIPTION
**Pull request by davemooreuws**
closes #194

### CODE REVIEW

**Review by tjholm — APPROVED** [review]
LGTM

File: packages/common/src/stack/function.ts [hunk 1]
```diff
@@ -58,6 +58,14 @@ export class StackFunction<FuncExtensions = Record<string, any>> {
 		return this.name;
 	}
 
+	/**
+	 *
+	 * @returns the stack version of the function, else the stack version
```
Comment (jyecusch) [inline packages/common/src/stack/function.ts]: The comment seems a little confusing, isn't this returning the Membrane/framework version of the function? A slight update would be useful

File: packages/common/src/stack/stack.ts [hunk 2]
```diff
@@ -129,6 +129,14 @@ export class Stack<
 		this.descriptor.name = name;
 	}
 
+	/**
+	 *
+	 * @returns the stack version
```
Comment (jyecusch) [inline packages/common/src/stack/stack.ts]: A small update would add some clarity here, something like: 
`@returns the version of the nitric framework used by this stack`

File: packages/common/src/types/stack.ts [hunk 3]
```diff
@@ -37,6 +37,8 @@ export interface NitricStack<
 > {
 	// Name of the Nitric Stack
 	name: string;
+	// Version of the Nitric Stack
```
Comment (jyecusch) [inline packages/common/src/types/stack.ts]: I think this should be described as the Nitric Framework Version of the Stack

**Review by jyecusch — CHANGES_REQUESTED** [review]
Looks good, I've just suggested a few updates to the comments that could help with clarity

**Review by jyecusch — APPROVED** [review]

**Pull request merged**